backend/controllers/user_controller.js

```diff
@@ -2,12 +2,13 @@ const db = require("../models");
 const user = db.user;
 
 exports.post = async(req, res) => {
-    if (req.body.data.ingredients && req.body.data.equipment && req.body.data.restrictions) {
+    if (req.body.data.ingredients && req.body.data.equipment && req.body.data.restrictions && req.body.data.recipes) {
```

Comment:
Seems this should be an OR conditions rather than an AND condition. It could be that only one of ingredients, equipment or restrictions is getting updated.